Broken/Missing interaction with Brigadier API on Paper 1.20.6

When reloading a plugin (reloaded mcMMO as an example):
```
[13:03:08 WARN]: java.lang.NoSuchFieldException: vanillaCommandDispatcher
[13:03:08 WARN]:        at java.base/java.lang.Class.getDeclaredField(Class.java:2782)
[13:03:08 WARN]:        at PlugManX.jar//com.rylinaux.plugman.util.BukkitCommandWrap.resolveVanillaCommandDispatcherField(BukkitCommandWrap.java:212)
[13:03:08 WARN]:        at PlugManX.jar//com.rylinaux.plugman.util.BukkitCommandWrap.wrap(BukkitCommandWrap.java:68)
[13:03:08 WARN]:        at PlugManX.jar//com.rylinaux.plugman.pluginmanager.BukkitPluginManager.loadCommands(BukkitPluginManager.java:671)
[13:03:08 WARN]:        at PlugManX.jar//com.rylinaux.plugman.pluginmanager.PaperPluginManager.loadCommands(PaperPluginManager.java:444)
[13:03:08 WARN]:        at PlugManX.jar//com.rylinaux.plugman.pluginmanager.PaperPluginManager.lambda$load$2(PaperPluginManager.java:434)
[13:03:08 WARN]:        at org.bukkit.craftbukkit.v1_20_R4.scheduler.CraftTask.run(CraftTask.java:101)
[13:03:08 WARN]:        at org.bukkit.craftbukkit.v1_20_R4.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:482)
[13:03:08 WARN]:        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1704)
[13:03:08 WARN]:        at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:484)
[13:03:08 WARN]:        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1576)
[13:03:08 WARN]:        at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1236)
[13:03:08 WARN]:        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:328)
[13:03:08 WARN]:        at java.base/java.lang.Thread.run(Thread.java:1583)
```

(the `vanillaCommandDispatcher` was removed in the `Brigadier-based-command-API.patch`)

When reloading a plugin that uses Paper's new Brigadier API:
```
org.bukkit.command.CommandException: Unhandled exception executing command 'plugman' in plugin PlugManX v2.4.0
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[paper-mojangapi-1.20.6-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigCommand.run(BukkitCommandNode.java:91) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73) ~[brigadier-1.2.9.jar:?]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:31) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.commands.execution.CommandQueueEntry.action(CommandQueueEntry.java:5) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:103) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:448) ~[?:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:355) ~[?:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:342) ~[?:?]
        at net.minecraft.commands.Commands.performCommand(Commands.java:337) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.performUnsignedChatCommand(ServerGamePacketListenerImpl.java:2202) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$14(ServerGamePacketListenerImpl.java:2176) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:151) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1516) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.server.MinecraftServer.wrapRunnable(MinecraftServer.java:195) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:125) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1493) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1416) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:135) ~[?:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1382) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1243) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:328) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NullPointerException
        at java.base/java.util.Objects.requireNonNull(Objects.java:233) ~[?:?]
        at io.papermc.paper.command.brigadier.PluginCommandNode.getPlugin(PluginCommandNode.java:35) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at io.papermc.paper.command.brigadier.PaperBrigadier.wrapNode(PaperBrigadier.java:53) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at io.papermc.paper.command.brigadier.bukkit.BukkitBrigForwardingMap.nodeToEntry(BukkitBrigForwardingMap.java:308) ~[sparklypaper-1.20.6.jar:1.20.6-DEV-a8ce987]
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
        at com.google.common.collect.CollectSpliterators$1.lambda$tryAdvance$0(CollectSpliterators.java:123) ~[guava-32.1.2-jre.jar:?]
        at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1950) ~[?:?]
        at com.google.common.collect.CollectSpliterators$1.tryAdvance(CollectSpliterators.java:122) ~[guava-32.1.2-jre.jar:?]
        at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292) ~[?:?]
        at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206) ~[?:?]
        at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169) ~[?:?]
        at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298) ~[?:?]
        at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681) ~[?:?]
        at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:337) ~[?:?]
        at java.base/java.util.HashSet.<init>(HashSet.java:121) ~[?:?]
        at PlugManX.jar/com.rylinaux.plugman.pluginmanager.ModernPaperPluginManager.unload(ModernPaperPluginManager.java:97) ~[PlugManX.jar:?]
        at PlugManX.jar/com.rylinaux.plugman.pluginmanager.PaperPluginManager.reload(PaperPluginManager.java:469) ~[PlugManX.jar:?]
        at PlugManX.jar/com.rylinaux.plugman.command.ReloadCommand.execute(ReloadCommand.java:125) ~[PlugManX.jar:?]
        at PlugManX.jar/com.rylinaux.plugman.PlugManCommandHandler.onCommand(PlugManCommandHandler.java:94) ~[PlugManX.jar:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[paper-mojangapi-1.20.6-R0.1-SNAPSHOT.jar:?]
        ... 26 more
```

And yes, I'm using the latest version

```
> plugman info PlugManX
[13:04:58 INFO]: [PlugMan] Plugin Information: PlugManX
[13:04:58 INFO]: - Version: 2.4.0
[13:04:58 INFO]: - Author(s): rylinaux, Entity303
[13:04:58 INFO]: - Status: Enabled
[13:04:58 INFO]: - SoftDepends: PlugMan
```

> Changed the title to say `SparklyPaper` instead of `Paper` 'till I am able to reproduce this on pure paper, or someone else provides me with the same error on pure paper.
> 
> Please note that PlugManX may react differently to paper forks than on pure paper

[SparklyPaper is my own Paper fork](https://github.com/SparklyPower/SparklyPaper/tree/ver/1.20.6/patches/server), and anything related to Brigadier commands

But it is still my fault, the mcMMO stacktrace issue happened because I was using a reobfuscated server JAR instead of using the mojmapped JAR (whoops)

But reloading plugins that use the new new Brigadier API still breaks, here's the stacktraces in Paper 1.20.6 `1.20.6-130-master@5d8e53d`

```
[10:06:19 WARN]: Unexpected exception while parsing console command "plugman reload DreamSonecas"
org.bukkit.command.CommandException: Unhandled exception executing command 'plugman' in plugin PlugManX v2.4.0
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[paper-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:164) ~[paper-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.CraftServer.dispatchCommand(CraftServer.java:964) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at org.bukkit.craftbukkit.CraftServer.dispatchServerCommand(CraftServer.java:949) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at net.minecraft.server.dedicated.DedicatedServer.handleConsoleInputs(DedicatedServer.java:521) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:468) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1571) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1231) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:323) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NullPointerException
        at java.base/java.util.Objects.requireNonNull(Objects.java:233) ~[?:?]
        at io.papermc.paper.command.brigadier.PluginCommandNode.getPlugin(PluginCommandNode.java:35) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at io.papermc.paper.command.brigadier.PaperBrigadier.wrapNode(PaperBrigadier.java:53) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at io.papermc.paper.command.brigadier.bukkit.BukkitBrigForwardingMap.nodeToEntry(BukkitBrigForwardingMap.java:308) ~[paper-1.20.6.jar:1.20.6-130-5d8e53d]
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
        at com.google.common.collect.CollectSpliterators$1.lambda$tryAdvance$0(CollectSpliterators.java:123) ~[guava-32.1.2-jre.jar:?]
        at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1950) ~[?:?]
        at com.google.common.collect.CollectSpliterators$1.tryAdvance(CollectSpliterators.java:122) ~[guava-32.1.2-jre.jar:?]
        at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292) ~[?:?]
        at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206) ~[?:?]
        at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169) ~[?:?]
        at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298) ~[?:?]
        at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681) ~[?:?]
        at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:337) ~[?:?]
        at java.base/java.util.HashSet.<init>(HashSet.java:121) ~[?:?]
        at PlugManX.jar/com.rylinaux.plugman.pluginmanager.ModernPaperPluginManager.unload(ModernPaperPluginManager.java:97) ~[PlugManX.jar:?]
        at PlugManX.jar/com.rylinaux.plugman.pluginmanager.PaperPluginManager.reload(PaperPluginManager.java:469) ~[PlugManX.jar:?]
        at PlugManX.jar/com.rylinaux.plugman.command.ReloadCommand.execute(ReloadCommand.java:125) ~[PlugManX.jar:?]
        at PlugManX.jar/com.rylinaux.plugman.PlugManCommandHandler.onCommand(PlugManCommandHandler.java:94) ~[PlugManX.jar:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[paper-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        ... 9 more
```

--- a/src/main/java/com/rylinaux/plugman/pluginmanager/ModernPaperPluginManager.java
+++ b/src/main/java/com/rylinaux/plugman/pluginmanager/ModernPaperPluginManager.java
@@ -38,6 +38,8 @@ public class ModernPaperPluginManager extends PaperPluginManager {
             Map<String, Plugin> names;
             Map<String, Command> commands;
             Map<Event, SortedSet<RegisteredListener>> listeners = null;
+            Map<String, Object> lookupNames;
+            List<Plugin> pluginList;
             boolean reloadlisteners = true;
 
             pluginManager.disablePlugin(plugin);
@@ -51,14 +53,12 @@ public class ModernPaperPluginManager extends PaperPluginManager {
 
                 Field lookupNamesField = instanceManager.getClass().getDeclaredField("lookupNames");
                 lookupNamesField.setAccessible(true);
-                Map<String, Object> lookupNames = (Map<String, Object>) lookupNamesField.get(instanceManager);
+                lookupNames = (Map<String, Object>) lookupNamesField.get(instanceManager);
 
-                lookupNames.remove(plugin.getName().toLowerCase());
 
                 Field pluginsField = instanceManager.getClass().getDeclaredField("plugins");
                 pluginsField.setAccessible(true);
-                List<Plugin> pluginList = (List<Plugin>) pluginsField.get(instanceManager);
-                pluginList.remove(plugin);
+                pluginList = (List<Plugin>) pluginsField.get(instanceManager);
 
                 pluginsField = Bukkit.getPluginManager().getClass().getDeclaredField("plugins");
                 pluginsField.setAccessible(true);
@@ -131,6 +131,11 @@ public class ModernPaperPluginManager extends PaperPluginManager {
                         }
                     }
                 }
+
+            // The plugin can only be removed from the lookup names and the plugin list AFTER the commands are unregistered, to avoid issues with commands created via Paper's Brigadier API
+            lookupNames.remove(plugin.getName().toLowerCase());
+            pluginList.remove(plugin);
+
             if (plugins != null)
                 plugins.remove(plugin);
             if (names != null)

